RCS file: /var/cvsup/gsl/fpicker/source/unx/gnome SalGtkFilePicker.cxx,v
retrieving revision 1.7
retrieving revision 1.7.16.1
diff -u -r1.7 -r1.7.16.1
--- fpicker/source/unx/gnome/SalGtkFilePicker.cxx	2005/07/12 11:58:49	1.7
+++ fpicker/source/unx/gnome/SalGtkFilePicker.cxx	2005/08/25 12:35:02	1.7.16.1
@@ -200,7 +200,9 @@
     mbPreviewState( sal_False ),
 	m_pPreview( NULL ),
 	m_PreviewImageWidth( 256 ),
-	m_PreviewImageHeight( 256 )
+	m_PreviewImageHeight( 256 ),
+    mnHID_FolderChange( 0 ), 
+    mnHID_SelectionChange( 0 )
 {	
 	int i;
 
@@ -785,17 +787,13 @@
     }
 }
 
-rtl::OUString SAL_CALL SalGtkFilePicker::getCurrentFilter() throw( uno::RuntimeException )
+void SalGtkFilePicker::UpdateFilterfromUI()
 {
-	OSL_ASSERT( m_pDialog != NULL );
-	::vos::OGuard aGuard( Application::GetSolarMutex() );
-	// TODO return m_pImpl->getCurrentFilter();
-
-	OSL_TRACE( "GetCURRENTfilter\n" );
-
 	// Update the filtername from the users selection if they have had a chance to do so.
     // If the user explicitly sets a type then use that, if not then take the implicit type
     // from the filter of the files glob on which he is currently searching
+	if (!mnHID_FolderChange	|| !mnHID_SelectionChange)
+        return;
     GtkTreeSelection* selection = gtk_tree_view_get_selection(GTK_TREE_VIEW(m_pFilterView));
     GtkTreeIter iter;
     GtkTreeModel *model;
@@ -810,6 +808,16 @@
     {
         updateCurrentFilterFromName(gtk_file_filter_get_name( filter ));
     }
+}
+
+rtl::OUString SAL_CALL SalGtkFilePicker::getCurrentFilter() throw( uno::RuntimeException )
+{
+	OSL_ASSERT( m_pDialog != NULL );
+	::vos::OGuard aGuard( Application::GetSolarMutex() );
+
+	OSL_TRACE( "GetCURRENTfilter\n" );
+
+    UpdateFilterfromUI();
 
 	OSL_TRACE( "Returning current filter of %s\n", 
 		OUStringToOString( m_aCurrentFilter, RTL_TEXTENCODING_UTF8 ).getStr() );
@@ -1021,16 +1029,16 @@
 
 	sal_Int16 retVal = 0;
 
-	gulong nHID_FolderChange = 
+	SetFilters();
+
+	mnHID_FolderChange = 
 		g_signal_connect( GTK_FILE_CHOOSER( m_pDialog ), "current-folder-changed",
 			G_CALLBACK( folder_changed_cb ), ( gpointer )this );
 
-	gulong nHID_SelectionChange = 
+	mnHID_SelectionChange = 
 		g_signal_connect( GTK_FILE_CHOOSER( m_pDialog ), "selection-changed",
 			G_CALLBACK( selection_changed_cb ), ( gpointer )this );
 
-	SetFilters();
-
 	int btn = GTK_RESPONSE_NO;
 	int nRes = GTK_RESPONSE_CANCEL;
 	
@@ -1088,10 +1096,10 @@
 		}
 	}
     
-	if (nHID_FolderChange)
-		g_signal_handler_disconnect(GTK_FILE_CHOOSER( m_pDialog ), nHID_FolderChange);
-	if (nHID_SelectionChange)
-		g_signal_handler_disconnect(GTK_FILE_CHOOSER( m_pDialog ), nHID_SelectionChange);
+	if (mnHID_FolderChange)
+		g_signal_handler_disconnect(GTK_FILE_CHOOSER( m_pDialog ), mnHID_FolderChange);
+	if (mnHID_SelectionChange)
+		g_signal_handler_disconnect(GTK_FILE_CHOOSER( m_pDialog ), mnHID_SelectionChange);
     
 	return retVal;
 }

RCS file: /var/cvsup/gsl/fpicker/source/unx/gnome SalGtkFilePicker.hxx,v
retrieving revision 1.5
retrieving revision 1.5.18.1
diff -u -r1.5 -r1.5.18.1
--- fpicker/source/unx/gnome/SalGtkFilePicker.hxx	2005/07/12 11:59:09	1.5
+++ fpicker/source/unx/gnome/SalGtkFilePicker.hxx	2005/08/25 12:35:02	1.5.18.1
@@ -357,6 +357,8 @@
 		GtkWidget *m_pLists[ LIST_LAST ];
 		GtkWidget *m_pListLabels[ LIST_LAST ];
 		bool mbListVisibility[ LIST_LAST ];
+        gulong mnHID_FolderChange;
+        gulong mnHID_SelectionChange;
 
 		::rtl::OUString	m_aCurrentFilter;
 
@@ -364,6 +366,7 @@
 
 		void SetCurFilter( const OUString&amp; rFilter );
 		void SetFilters();
+        void UpdateFilterfromUI();
 
         void implChangeType( GtkTreeSelection *selection );
 		int implAddFilter( const OUString&amp; rFilter, const OUString&amp; rType);
